test/integration-test.py

```diff
@@ -38,8 +39,18 @@ def run_client_tests():
     cmd = os.path.join(root, 'tests', 'boulder-integration.sh')
     run(cmd, cwd=root)
 
-def run_go_tests():
-    run("go test -tags integration -count=1 ./test/integration")
+def run_go_tests(filterPattern=None):
+    """
+    run_go_tests launches the Go integration tests. The go test command must
+    return zero or an exception will be raised. If the filterPattern is provided
+    it is used as the value of the `--test.run` argument to the go test command.
+    """
+    cmdLine = "go test"
+    if filterPattern is not None and filterPattern != "":
+        cmdLine = "{0} --test.run {1}".format(cmdLine, filterPattern)
+    cmdLine = "{0} -tags integration -count=1 ./test/integration".format(cmdLine)
+    cmd = shlex.split(cmdLine)
```

Comment:
This will do the wrong thing if the filterPattern has spaces in it. I'd recommend building up the args in an array:

["go", "test", "--test.run", filterPattern, "-tags=integration" ...]

Reply:
I think it's a theoretical concern given you can't have a space in the name of a Python test function or a Go test function but I don't feel strongly. Fixed in 7caf233